### PR TITLE
adds poisonous snakes.

### PR DESCRIPTION
**Pull request by Arthurdentist**
This PR aims to add poisonous snakes. It is not ready to be merged. Most of the work is done. Things to do.
1. Sort out snake meat so Snake Venom can be obtained from it.
2. Make Anti-Venom work properly.
3. Finish up descriptions on snakes and other things.
4. Fix anything I missed.
5. Balance the shit out of everything.

Poisonous snakes are weak as fuck. They have 10 health and can be killed in a single laser shot. Their strength lies in their venom which can weaken you. If you find yourself surrounded by a lot of snakes then you'll end up getting weakened over and over, preventing you from escaping. Death from a thousand cuts pretty much.

poison snake

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/snake.dm [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	var/poison_type = SVENOM
+
+
+/mob/living/simple_animal/hostile/snake/AttackingTarget()
+	..()
+	if(isliving(target))
+		var/mob/living/L = target
+		if(L.reagents)
+			if(prob(poison_chance))
+				src.visible_message("<span class='warning'>\the [src] injects a powerful toxin into [L]!</span>")
```
Comment (PJB3005) [inline code/modules/mob/living/simple_animal/hostile/snake.dm]: ES AR SEE DOT

File: code/modules/mob/living/simple_animal/hostile/snake.dm [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+				L.reagents.add_reagent(poison_type, poison_per_bite)
```
Comment (PJB3005) [inline code/modules/mob/living/simple_animal/hostile/snake.dm:39]: Pretty sure there's probably a helper proc to wrap this so you don't have to manually call the proc.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 3]
```diff
@@ -501,6 +501,8 @@
 		holder.remove_reagent("zombiepowder", 0.5 * REM)
 	if(holder.has_reagent("mindbreaker"))
 		holder.remove_reagent("mindbreaker", 2 * REM)
+	if(holder.has_reagent("svenom"))
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Reagents.dm]: Use the defines, please.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 4]
```diff
@@ -5282,3 +5284,39 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
...
+	id = AVENOM
+	description = "A strong serum used to neutralise snake venom."
+	reagent_state = LIQUID
+	color = "#004A7F"
+
+datum/reagent/antivenom/on_mob_life(var/mob/living/M as mob)
+	if(..())
+		return 1
+
+	if(holder.has_reagent("svenom"))
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Reagents.dm]: Again, defines.

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 5]
```diff
@@ -2698,6 +2698,12 @@
 	required_reagents = list(MINDBREAKER = 1, OXYGEN = 1, INAPROVALINE = 1)
 	result_amount = 3
 
+/datum/chemical_reaction/antivenom
+	name = "Anti venom"
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Recipes.dm]: Shouldn't this have a hyphen?

File: code/modules/mob/living/simple_animal/hostile/snake.dm [hunk 6]
```diff
@@ -0,0 +1,38 @@
+/mob/living/simple_animal/hostile/snake
+	name = "snake"
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/snake.dm:2]: What's the description?

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 7]
```diff
@@ -5282,3 +5284,39 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
...
+	description = "Venom straight from the deadliest of space snakes."
+	reagent_state = LIQUID
+	color = "#055400"
+
+datum/reagent/snakevenom/on_mob_life(var/mob/living/M as mob)
+	if(..())
+		return 1
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(H.species.name == "Unathi") //filthy lizards are immune to snake venom
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Reagents.dm]: Make this `if(!isunathi(H))`, no need for the empty if like that.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 8]
```diff
@@ -5282,3 +5284,39 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
...
+			return
+		else
+			M.adjustToxLoss(1)
+			if(prob(25))
+				M.AdjustWeakened(2)
+
+datum/reagent/antivenom
+	name = "Anti Venom"
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Reagents.dm]: Hyphen, I believe.

File: code/modules/mob/living/simple_animal/hostile/snake.dm (hunk 6 above)
Comment (Arthurdentist) [reply code/modules/mob/living/simple_animal/hostile/snake.dm:2]: going to be sorted out tomorrow

**Comment by ghost** [comment]
So, how do you get the snake venom to make the antivenom?

Maybe have it have some butcherable teeth that can then be ground down into snake venom?

**Comment by Arthurdentist** [comment]
That is the idea Skullyton.

Basically you want to keep away from the snakes and deal with them at a distance. They don't aggro unless you get close so you can safely kill them from afar.

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 7 above)
Comment (ghost) [inline code/modules/reagents/Chemistry-Reagents.dm]: This PROBABLY should be a species variable or flag or something

**Comment by Exxion** [comment]
God damn it
They're _venomous_ if they inject venom when they bite you
Poisonous just means they're toxic to eat

**Comment by Arthurdentist** [comment]
fancy eating one to see how it goes?

**Comment by Exxion** [comment]
They might be poisonous too, but venomous is the important part here

**Comment by Arthurdentist** [comment]
I'll change it tomorrow. 

I personally get triggered by people playing security saying 'Grand Trespass' despite it not being a real thing so I can understand the aggravation.

File: code/modules/mob/living/simple_animal/hostile/snake.dm (hunk 1 above)
Comment (ComicIronic) [reply code/modules/mob/living/simple_animal/hostile/snake.dm]: NON
STANDARD
FORMATTING
MEME
Seriously, stop posting these until it's in the rules, and it's not in the rules.

**Comment by CptWad** [comment]
just have a venom sac that you can bucher out of it
also snake skin

**Comment by Arthurdentist** [comment]
wew merge conflicts

**Comment by NigglyWiggly** [comment]
I made a snake sprite based off this cute boy, if you want it.



I think I also made the eastern coral snake, but I'd have to check. wouldn't be hard to throw together

**Comment by Arthurdentist** [comment]
I fucking hate trying to resolve merge conflicts, closing this and pretending it never happened.

**Comment by CptWad** [comment]
r i p

**Comment by ConkerDev** [comment]
Snake, what happened? Snake!?  SNAAAAAAAAAAAKEEE!

**Pull request closed without merging**